### DISCUSSION TITLE
why my st2 workflows always assign null value on action output even my action can give the output?

### DISCUSSION THREAD

**Discussion by idham1311** [StackStorm/st2#5807] [Workflows] [answered]
## SUMMARY
Hello, I already create the sample workflow to make sure that I can get the output from action output. But when the workflow run successfully. It doesnt show any output. Anyone know how to set the value of the output?

### STACKSTORM VERSION
st2 3.7.0, on Python 3.8.10

##### OS, environment, install method
linux

## Steps to reproduce the problem
Show how to reproduce the problem, using a minimal test-case. Make sure to include any content
(pack content - workflows, actions, etc.) which are needed to reproduce the problem.
this is my workflow:
![image](https://user-images.githubusercontent.com/43712404/201885552-97347e86-aa2f-4ecd-aed8-096649e65fd1.png)
this is result for mysqldump:
![image](https://user-images.githubusercontent.com/43712404/201885807-0cc54e8c-b6f4-48ff-9615-51d872f49121.png)

## Expected Results
expect action output can shows the output from the action on the workflows.
What did you expect to happen when running the steps above?

## Actual Results
![image](https://user-images.githubusercontent.com/43712404/201885108-a238c19e-4be9-4b2c-802b-6902e5ee8622.png)

What happened? What output did you get?
null values

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by arm4b** [comment]
Converted to a Discussion.
Could you provide your workflow example so someone answering your question can look in more detail?

**Reply by idham1311** [reply]
it's okay, I already find the solution. Thank you!!

**Comment by fdrab** [comment, marked as answer]
Have you assigned the output of the action to the output of the workflow? As far as I can tell, there's no "drag and drop" way to do this, you have to edit the yaml file (in the workflow designer. Consider my workflow that contains a single operation that outputs and publishes date into "outputs" variable:
![Screenshot 2022-11-15 120227](https://user-images.githubusercontent.com/118189175/201908470-440898fe-5bf5-43ad-a282-8788f8a1f57e.png)

This is the output of the single action. If I want the output of said action as the output of the workflow itself, I have to assign it to the workflow output in the yaml definition like this:

![Screenshot 2022-11-15 120255](https://user-images.githubusercontent.com/118189175/201908618-2b8ba4d8-b951-4089-8bf4-902e68fd4ae3.png)

**Reply by idham1311** [reply]
Thank you!!
